apis/flowcollector/v1beta2/flowcollector_types.go

```diff
@@ -373,6 +373,80 @@ type FlowCollectorIPFIXReceiver struct {
 	Transport string `json:"transport,omitempty"`
 }
 
+type FlowCollectorOpenTelemetryLogs struct {
+	// Set `enable` to `true` to send logs to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+}
+
+type FlowCollectorOpenTelemetryMetrics struct {
+	// Set `enable` to `true` to send metrics to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+
+	// How often should metrics be sent to collector
+	// +kubebuilder:default:="20s"
+	PushTimeInterval *metav1.Duration `json:"pushTimeInterval,omitempty"`
+}
+
+type FlowCollectorOpenTelemetryTraces struct {
+	// Set `enable` to `true` to send traces to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+
+	// Separate span for each prefix listed
+	//+optional
+	SpanSplitter []string `json:"spanSplitter,omitempty"`
+}
+
+type GenericTransformRule struct {
+	Input      string `json:"input,omitempty"`
+	Output     string `json:"output,omitempty"`
+	Multiplier int    `json:"multiplier,omitempty"`
+}
+
+type GenericTransform []GenericTransformRule
+
+type FlowCollectorOpenTelemetry struct {
+	// Address of the Open Telemetry receiver
+	// +kubebuilder:default:=""
+	TargetHost string `json:"targetHost"`
+
+	// Port for the Open Telemetry receiver
+	TargetPort int `json:"targetPort"`
+
+	// Protocol of Open Telemetry connection. The available options are `http` and `grpc`.
+	// +optional
+	Protocol string `json:"protocol"`
+
+	// Headers to add to messages (optional)
+	// +optional
+	Headers map[string]string `json:"headers,omitempty"`
+
+	// TLS client configuration.
+	// +optional
+	TLS ClientTLS `json:"tls"`
+
+	// Custom rules to transform flow logs to Opentelemetry-conformant format.
+	// By default the following format will be used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal
+	// Setting `customRules` will fully override the default rules allowing you to export each field to your naming convention
+	// +optional
+	Rules *[]GenericTransformRule `json:"customRules,omitempty"`
```

Comment:
In general we try to make the CRD more opinionated than what FLP exposes. Here, I think we should present that more clearly as a mapping of fields, and tell why we expose it (because there's no standard)

Here's a proposal:

```suggestion
	// Custom fields mapping to an OpenTelemetry conformant format. 
	// By default, NetObserv format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal .
	// As there is currently no accepted otlp standard for L3/4 network logs, you can freely override it with your own.
	// +optional
	FieldsMapping *[]GenericTransformRule `json:"fieldsMapping,omitempty"`
```

Reply:
also done in https://github.com/netobserv/network-observability-operator/pull/671/commits/efdb0d4e59cf584864ef8acc96cdf6fdf2b3f8d7